src/Azure.Functions.Cli/Kubernetes/FuncKeys/FuncAppKeysHelper.cs

```diff
@@ -0,0 +1,118 @@
+        private const string KubernetesSecretsMountPath = "/run/secrets/functions-keys";
+        private const string AzureWebJobsSecretStorageTypeEnvVariableName = "AzureWebJobsSecretStorageType";
+        private const string AzureWebJobsKubernetesSecretNameEnvVariableName = "AzureWebJobsKubernetesSecretName";
+        private const string MasterKey = "host.master";
+        private const string HostFunctionKey = "host.function.default";
+        private const string HostSystemKey = "host.systemKey.default";
+        private const string FunctionKeyPrefix = "functions.";
+        private const string FunctionDefaultKeyName = "default";
+        /// <summary>
+        /// Implementation of this method creates the Host and Function Keys
+        /// </summary>
+        /// <param name="functionNames">The <see cref="IEnumerable{string}"></see> of function names </param>
+        /// <returns>The <see cref="IDictionary{string, string}"/> of function app's host and function keys</returns>
+        public static IDictionary<string, string> CreateKeys(IEnumerable<string> functionNames)
+        {
+            var funcAppKeys = new Dictionary<string, string>
+            {
+                { MasterKey, GenerateKey() },
+                { HostFunctionKey, GenerateKey() },
+                { HostSystemKey, GenerateKey() }
+            };
+
+            if (functionNames?.Any() == true)
+            {
+                foreach (var funcName in functionNames)
+                {
+                    funcAppKeys[$"{FunctionKeyPrefix}{funcName}.{FunctionDefaultKeyName}"] = GenerateKey();
+                }
+            }
+
+            return funcAppKeys;
+        }
+
+        public static void AddAppKeysEnvironVariableNames(IDictionary<string, string> envVariables,
+            string keysSecretCollectionName,
+            bool mountKeysAsContainerVolume)
+        {
+            if (envVariables == null)
+            {
+                envVariables = new Dictionary<string, string>();
+            }
+
+            //if keysSecretCollectionName, mountKeysAsContainerVolume has been passed as deployment parameter then that means the func app keys needs to be managed as kubernetes secret
+            if ((!string.IsNullOrWhiteSpace(keysSecretCollectionName) || mountKeysAsContainerVolume)
+                && !envVariables.ContainsKey(AzureWebJobsSecretStorageTypeEnvVariableName))
+            {
+                envVariables.Add(AzureWebJobsSecretStorageTypeEnvVariableName, "kubernetes");
+            }
+
+            if (!envVariables.ContainsKey(AzureWebJobsKubernetesSecretNameEnvVariableName)
+                && !mountKeysAsContainerVolume)
+            {
+                if (!string.IsNullOrWhiteSpace(keysSecretCollectionName))
+                {
+                    envVariables.Add(AzureWebJobsKubernetesSecretNameEnvVariableName, $"secrets/{keysSecretCollectionName}");
+                }
+            }
```

Comment:
I think the logic of this method can be significantly simplified if you just default the `keysSecretCollectionName` in `GetFunctionsDeploymentResources()` to a value if it's null. That way you just need to check `mountKeysAsContainerVolume` in one if statement, right?